pkg/clientcert/certificate.go

```diff
@@ -157,16 +166,81 @@ func BuildKubeconfig(clientConfig *restclient.Config, certPath, keyPath string)
 	return kubeconfig
 }
 
-// isCSRApproved returns true if the given csr has been approved
-func isCSRApproved(csr *certificatesv1.CertificateSigningRequest) bool {
+type csrControl interface {
+	get(name string) (metav1.Object, error)
+	create(ctx context.Context, recorder events.Recorder, objMeta metav1.ObjectMeta, csrData []byte, signerName string) (string, error)
+	isApproved(csr metav1.Object) bool
+	isIssued(csr metav1.Object) ([]byte, bool)
```

Comment:
it seems that we get the certificate  from csr status in both v1 and v1beta1, what the different is between them? we may do not need this